src/celeritas/field/DormandPrinceMultiStepperGlobal.cuda.hh

```diff
@@ -0,0 +1,280 @@
+/*!
+ * Based on the DormandPrinceStepper.hh, but with multiple threads.
+ */
+template<class EquationT>
+class DormandPrinceMultiStepperGlobal
+{
+  public:
+    //!@{
+    //! \name Type aliases
+    using result_type = FieldStepperResult;
+    //!@}
+
+  public:
+    //! Construct with the equation of motion
+    explicit CELER_FUNCTION DormandPrinceMultiStepperGlobal(EquationT&& eq)
+        : calc_rhs_(::celeritas::forward<EquationT>(eq))
+    {
+    }
+
+    // Adaptive step size control
+    CELER_FUNCTION result_type operator()(real_type step,
+                                          OdeState const& beg_state,
+                                          int number_threads,
+                                          OdeState* ks,
+                                          OdeState* along_state,
+                                          FieldStepperResult* result) const;
+
+    CELER_FUNCTION void run_sequential(real_type step,
+                                       OdeState const& beg_state,
+                                       int id,
+                                       int mask,
+                                       OdeState* ks,
+                                       OdeState* along_state,
+                                       FieldStepperResult* result) const;
+
+    CELER_FUNCTION void run_aside(real_type step,
+                                  OdeState const& beg_state,
+                                  int id,
+                                  int index,
+                                  int mask,
+                                  OdeState* ks,
+                                  OdeState* along_state,
+                                  FieldStepperResult* result) const;
+
+  private:
+    // Functor to calculate the force applied to a particle
+    EquationT calc_rhs_;
+};
+
+//---------------------------------------------------------------------------//
+// MACROS
+//---------------------------------------------------------------------------//
+#define UPDATE_STATE(index, state, coefficient, k)        \
+    state.pos[index - 1] = coefficient * k.pos[index - 1] \
+                           + state.pos[index - 1];        \
+    state.mom[index - 1] = coefficient * k.mom[index - 1] \
+                           + state.mom[index - 1];
+
+#define DISPATCH_VECT_MULT(mask) \
+    __syncwarp(mask);            \
+    __syncwarp(mask);
+
+//---------------------------------------------------------------------------//
+// DEDUCTION GUIDES
+//---------------------------------------------------------------------------//
+template<class EquationT>
+CELER_FUNCTION DormandPrinceMultiStepperGlobal(EquationT&&)
+    ->DormandPrinceMultiStepperGlobal<EquationT>;
+
+//---------------------------------------------------------------------------//
+// INLINE FUNCTIONS
+//---------------------------------------------------------------------------//
+#if !CELER_USE_DEVICE
+
+// TODO: implement this for CPU
```

Comment:
Is there even a CPU implementation that makes sense here? I'm not sure it's worth trying to implement the same algorithm for CPU since threads are limited.